service/history/timerBuilder.go

```diff
@@ -280,17 +289,21 @@ func (tb *timerBuilder) loadActivityTimers(msBuilder *mutableStateBuilder) {
 					EventID:     v.StartedID,
 					TimeoutType: w.TimeoutType_START_TO_CLOSE,
 					TimeoutSec:  v.StartToCloseTimeout,
-					TaskCreated: v.TimerTaskStatus == TimerTaskStatusCreated}
+					TaskCreated: (v.TimerTaskStatus & TimerTaskStatusCreatedStartToClose) != 0}
```

Comment:
Is it possible for the same timer to represent two different timeouts i.e say, both startToClose and scheduleToStart ? 

If not, the bitwise AND operation is un-necessary, just do a simple equality check, which is more readable.

If the answer is yes, can you create a type for TimerTaskStatus and add a method called Equals() that does this bit check ? Then your code will look like `v.TimerTaskStatus.Equals(TimerTaskStatusCreatedStartToClose)`